sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuiteBase.scala

```diff
@@ -1038,6 +1034,13 @@ abstract class CastSuiteBase extends SparkFunSuite with ExpressionEvalHelper {
           }.getMessage
           assert(e.contains("Error parsing interval year-month string: integer overflow"))
         }
+
+      Seq("1-1", " 1-8").foreach { interval =>
+          val e = intercept[IllegalArgumentException] {
+            cast(Literal.create(interval), YearMonthIntervalType()).eval()
+          }.getMessage
+          assert(e.contains("Interval string does not match year-month format"))
```

Comment:
I'm confused. Why should this fail? This is a breaking change

Reply:
> I'm confused. Why should this fail? This is a breaking change

Hmmmmmm
![image](https://user-images.githubusercontent.com/46485123/123111968-eac26980-d46f-11eb-8ca5-c6181ed82538.png)

Reply:
and we should NOT fail if it's INTERVAL YEAR TO MONTH

Reply:
`YearMonthIntervalType()` is INTERVAL YEAR TO MONTH

Reply:
Updated